### PR TITLE
layout overhaul

### PR DESCRIPTION
**Pull request by sigmahour**
Todo: Update links in protobuf repo after merge:

- `/docs/developers/firmware/encryption` => `/docs/about/overview/encryption`

### CODE REVIEW

**Comment by sachaw** [comment]
Why has the protobuf API page been moved out of the developer docs

File: docs/about/overview/mesh-alg.mdx [hunk 1]
```diff
@@ -1,23 +1,24 @@
 ---
 id: mesh-alg
```
Comment (jfirwin) [inline docs/about/overview/mesh-alg.mdx:2]: My only comment on this file is in the future we may want to make this document more accessible to newcomers to the project.

File: docs/about/overview/range-test.mdx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
```
Comment (jfirwin) [inline docs/about/overview/range-test.mdx:1]: I think this is great! This type of content may eventually be moved to the showcase section. But I'm glad you pulled the data here.

File: docs/configuration/device-config/display.mdx [hunk 3]
```diff
@@ -1,6 +1,7 @@
 ---
 id: display
 title: Display Configuration
+slug: /settings/config/display
```
Comment (jfirwin) [inline docs/configuration/device-config/display.mdx:4]: Nice. Simple. Prevents future link moving. I love it.

File: docs/configuration/module-config/canned-message.mdx [hunk 4]
```diff
@@ -1,6 +1,7 @@
 ---
 id: canned-message
 title: Canned Message Module Configuration
+slug: /settings/moduleconfig/canned-message
```
Comment (jfirwin) [inline docs/configuration/module-config/canned-message.mdx:4]: On the `moduleconfig` slugs, should we keep it in line with the folder naming convention? `module-config`?

File: docs/development/documentation/index.mdx [hunk 5]
```diff
@@ -1,19 +1,20 @@
...
+All of our documentation resides on GitHub. Instructions for setting up your GitHub account are located [here](/docs/development/reference/github).
 
-Our documentation is powered by [Docusaurus](https://docusaurus.io) — a documentation platform built on React that utilizes markdown files. Because markdown files are easy to edit, most content changes should be fairly simple. Instructions for setting up your instance of Docusaurus are located [here](docusaurus).
+Our documentation is powered by [Docusaurus](https://docusaurus.io) — a documentation platform built on React that utilizes markdown files. Because markdown files are easy to edit, most content changes should be fairly simple.
 
-Another component that we use is [Vercel](https://vercel.com) — a platform for frontend frameworks and static sites. Instructions for setting up your instance of Vercel are located [here](vercel).
+Another component that we use is [Vercel](https://vercel.com) — a platform for frontend frameworks and static sites. Instructions for setting up your instance of Vercel are located [here](/docs/development/documentation/publish#publish-to-vercel).
 
 ## Documentation Organization
```
Comment (jfirwin) [inline docs/development/documentation/index.mdx:19]: As i mentioned on discord we might want to make some changes here to reflect this reorganizing of the docs.

File: protobuf.tmpl [hunk 6]
```diff
@@ -1,9 +1,13 @@
 ---
-id: api
+id: protobuf-api
```
Comment (jfirwin) [inline protobuf.tmpl:2]: I've done very little with the protobuf generation, so I'll differ to @sachaw to make sure these changes are working correctly.

**Review by jfirwin — COMMENTED** [review]
Overall I think this is great. We will have some URL changes, which might cause some stir, but I think given the 2.0 overhaul the docs were due for a little dust to settle. I'd like @sachaw to also review this PR given it's size and my unfamiliarity with many of the 2.0 changes.

File: docs/configuration/module-config/canned-message.mdx (hunk 4 above)
Comment (garthvh) [reply docs/configuration/module-config/canned-message.mdx:4]: > 

I don't really like the idea of breaking all the config links, the current slug matches the protobuf structure and has been linked all over the forum and discord recently.

Comment (jfirwin) [reply docs/configuration/module-config/canned-message.mdx:4]: Sounds good, we'll leave it as is. It's just a cosmetic change that has big implications.

If we do change things, I think the `settings/bluetooth` `settings/modules/telemetry` or similar would be ideal.

File: docs/development/documentation/index.mdx [hunk 7]
```diff
@@ -24,8 +24,8 @@ Another component that we use is [Vercel](https://vercel.com) — a platform for
...
-|     Hardware Details     |         `docs/hardware`          |                  Any hardware related content. Any time a user is attaching a peripheral accessory to their device. That includes 3d printed cases, antennas, buttons, chimes, rotary encoders, and screens.                  |
-|    Radio Mesh Details    |           `docs/mesh`            |         This section discusses everything relating to the Meshtastic mesh. Mesh health metrics will be discussed here as well as topics such as signal strength, range and anyting else pertaining to "over the air".         |
+|     Hardware Details     |         `docs/hardware`          |                  Any hardware related content. Any time a user is attaching a peripheral accessory to their device. That includes 3D printed cases, antennas, buttons, chimes, rotary encoders, and screens.                  |
+|    Radio Mesh Details    |           `docs/mesh`            |         This section discusses everything relating to the Meshtastic mesh. Mesh health metrics will be discussed here as well as topics such as signal strength, range and anything else pertaining to "over the air".         |
```
Comment (jfirwin) [inline docs/development/documentation/index.mdx:28]: We removed this section, correct?

Comment (jfirwin) [reply docs/development/documentation/index.mdx:28]: (line 28)

File: docs/development/documentation/index.mdx (hunk 5 above)
Comment (sachaw) [reply docs/development/documentation/index.mdx:19]: Agreed, I once I get one of our last hurdles out the way (protoc-gen-docs) we can have pr reviews built automatically

File: protobuf.tmpl (hunk 6 above)
Comment (sachaw) [reply protobuf.tmpl:2]: Yeah, that's fine, just make sure the placeholder page is updated accordingly

**Review by sachaw — APPROVED** [review]

**Pull request merged**